### PR TITLE
Fix Nullpointer Exception in DOTCFGVisualizer#visualizeBlockNode() when analysis is null

### PR DESCRIPTION
**Pull request by CharlesZ-Chen**
The `DOTCFGVisualizer` should also visualise a Control Flow Graph without analysis.

However, the `DOTCFGVisualizer#visualizeBlockNode()` method will throw a null pointer exception when `analysis` is null, because it will always try to visualize the `abstractValue` of a `node` even if there actually no `analysis` is provided.

I changed this method to first append `node` to the `sbBlock`, and then only append `asbtractValue` from `analysis` if the `analysis` is non-null.

In this PR I also make other two small related changes:

1. add /*@Nullable*/ annotation on the parameter `analysis` of method `generateDotNodes()`, because this method also should accept a nullable `analysis`.

2. add assertion that "`analysis` is non-null" in method `visualizeBlockTransferInput()`, because this method should only be called when `analysis` is non-null.

### CODE REVIEW

File: dataflow/src/org/checkerframework/dataflow/cfg/DOTCFGVisualizer.java [hunk 1]
```diff
@@ -397,12 +400,11 @@ public void visualizeBlockTransferInput(Block bb, Analysis<A, S, T> analysis) {
 
     @Override
     public void visualizeBlockNode(Node t, /*@Nullable*/ Analysis<A, S, T> analysis) {
-        A value = analysis.getValue(t);
-        String valueInfo = "";
-        if (value != null) {
-            valueInfo = "    > " + prepareString(value.toString());
+        this.sbBlock.append(prepareString(t.toString()) + "   [ " + prepareNodeType(t) + " ]");
+        A value = null;
+        if (analysis != null && (value = analysis.getValue(t)) != null) {
```
Comment (wmdietl) [inline dataflow/src/org/checkerframework/dataflow/cfg/DOTCFGVisualizer.java]: Don't assign to a variable in a condition. Split this up into multiple conditions.

**Comment by CharlesZ-Chen** [comment]
Already split it up.

**Comment by wmdietl** [comment]
https://github.com/typetools/checker-framework/pull/831/files/7be8f301055fc5ec16eb5e2d0ee3590ad4c42e69#r69820225

**Comment by CharlesZ-Chen** [comment]
should I make changes according to Mike's comments directly on opp/master branch? or should I make changes here?

**Comment by wmdietl** [comment]
Let's do it on master, then it automatically becomes visible in the typetools pull request.
I need to try first creating a branch here, to make PRs against typetools easier.

**Pull request merged**